mlflow/utils/model_utils.py

```diff
@@ -263,3 +264,38 @@ def _validate_pyfunc_model_config(model_config):
             "Values in the provided ``model_config`` are of an unsupported type. Only "
             "JSON-serializable data types can be provided as values."
         )
+
+
+def _validate_model_assignment_in_init(cls_init_method_source):
```

Comment:
@harupy i applied the `ast` suggestion from the previous review, but had to do a bit of extra stuff to get it to work. since we're inspecting `__init__`, the lines are indented which makes the parse fail. i put some logic here to strip a fixed amount of whitespace from each line, but it feels a little hacky.

another option is to pass the entire source code of the module the class is defined in and then walk the entire AST of the file, but the code gets kind of messy for that. let me know what you think!